index.js

```diff
@@ -379,3 +381,144 @@ function toArray (obj) {
 function flatten (arr) {
   return Array.prototype.concat.apply([], arr)
 }
+
+function filterVersions (opts, versions, cb) {
+  const includeVersions = opts.versions.include ?? opts.versions
+  const excludeVersions = opts.versions.exclude
+  const mode = opts.versions.mode
+
+  versions = versions.filter(function (version) {
+    return semver.satisfies(version, includeVersions) && !semver.satisfies(version, excludeVersions)
+  })
+
+  switch (mode) {
+    case undefined:
+      return cb(null, versions)
+    case 'latest-majors':
+      return cb(null, getLatestMajors(versions))
+    case 'latest-minors':
+      return cb(null, getLatestMinors(versions))
+    default: {
+      const result = mode.match(/^max-(?<max>\d+)(-(?<algo>evenly|random|latest|popular))?$/)
+      if (!result) return cb(new Error(`Unknown mode: ${mode}`))
```

Comment:
have you thought on being more specific on the error?
- prefix `max` is wrong
- `{N}` is not a number
- `algo` is none of `evenly|random|latest|popular`

Reply:
I would prefer not to as it would require adding a lot of additional logic in the function, which I don't think is that important as long as the documentation clearly states the usage.